com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -138,15 +156,29 @@ public override void OnNetworkSpawn()
                         case AnimatorControllerParameterType.Float:
                             var value = m_Animator.GetFloat(cacheParam.Hash);
                             UnsafeUtility.WriteArrayElement(cacheParam.Value, 0, value);
+                            // Track this parameter
+                            if (!m_ParameterStates.ContainsKey(i))
+                            {
+                                m_ParameterStates.Add(i, new ParameterState() { CurrentStateFloat = value, AnimatorControllerParameterType = AnimatorControllerParameterType.Float });
+                            }
                             break;
```

Comment:
I'm looking at 2 different approaches in one place here.
what's the problem you're trying to solve here that can't be achieved with existing `UnsafeUtility.WriteArrayElement` approach?
I just don't like seeing Dictionaries everywhere and being thrown at every problem we have.
I'm just curious whether this issue could be solved more simple or not.

Reply:
Basically the NetworkAnimator will only update an Animator's parameters the animation is going through a transition and/or the animator's layer has changed.  While this works for when you are transitioning or have changed the layer, it misses all other cases where you are just changing the Animator's parameter values (i.e. no longer transitioning and not changing the animator's layer).
This table was a quick way to detect if a parameter has changed and if so it does a full update.

Really, I should bring this back to draft and make some optimization adjustments where it will only write what has changed and not everything each time it serializes.

But to answer the question of what that is for:  It tracks the current parameter values to be able to determine if one has changed.

Reply:
See my comments below, I now see what you were talking about and completely removed the dictionary.